### PR TITLE
Rename module `pytensor.tensor.var` to `pytensor.tensor.variable`

### PR DESCRIPTION
**Pull request by purna135**
Fixes [#421](https://github.com/pymc-devs/pytensor/issues/421)
Rename `pytensor.tensor.var` to `pytensor.tensor.variable`

Question:
do we require implementing a custom import mechanism for the module `var` along with issuing a deprecation warning to maintain backward compatibility?

### CODE REVIEW

**Comment by ricardoV94** [comment]
> do we require implementing a custom import mechanism for the module var along with issuing a deprecation warning to maintain backward compatibility?

That would be great

File: pytensor/tensor/var.py [hunk 1]
```diff
@@ -1,1079 +1,11 @@
...
-from pytensor.configdefaults import config
-from pytensor.graph.basic import Constant, OptionalApplyType, Variable
-from pytensor.graph.utils import MetaType
-from pytensor.scalar import ComplexError, IntegerDivisionError
-from pytensor.tensor import _get_vector_length
-from pytensor.tensor.exceptions import AdvancedIndexingError
-from pytensor.tensor.type import TensorType
-from pytensor.tensor.type_other import NoneConst
-from pytensor.tensor.utils import hash_from_ndarray
+from pytensor.tensor.variable import *  # noqa
```
Comment (purna135) [inline pytensor/tensor/var.py:4]: I was thinking of writing a custom `MetaPathFinder`, `Loader` method to support the import of `var` but then realized we can simply keep `var.py` file with a deprecation warning.
I'm not sure why the mypy is failing, need your help to fix that.

Comment (ricardoV94) [reply pytensor/tensor/var.py:4]: Keeping var.py with redirection + warning sounds fine. 

For mypy you may need to add the new file and remove the old one from the "list of files known to fail"

Comment (ricardoV94) [reply pytensor/tensor/var.py:4]: https://github.com/pymc-devs/pytensor/blob/main/scripts/mypy-failing.txt

Comment (purna135) [reply pytensor/tensor/var.py:4]: Thank you, but why it's failing for `pytensor/tensor/nlinalg.py` ? do I also need to add this file to mypy-failing list?

Comment (ricardoV94) [reply pytensor/tensor/var.py:4]: @jessegrabowski can you take a look at the nlinalg mypy failure?

Comment (jessegrabowski) [reply pytensor/tensor/var.py:4]: It comes from [this line](https://github.com/pymc-devs/pytensor/blob/288a3f3481e7dfa6162e1b60a3909a8f67b29337/pytensor/tensor/nlinalg.py#L302), dating back to 2017. It wasn't failing before?

Comment (purna135) [reply pytensor/tensor/var.py:4]: Thank you @jessegrabowski I made this change and it's working fine now

**Review by ricardoV94 — APPROVED** [review]

**Comment by ricardoV94** [comment]
Thanks @purna135

**Pull request merged**